tunnel_routes.go

```diff
@@ -0,0 +1,186 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// TunnelRoute is the full record for a route.
+type TunnelRoute struct {
+	Network    string     `json:"network"`
+	TunnelID   string     `json:"tunnel_id"`
+	TunnelName string     `json:"tunnel_name"`
+	Comment    string     `json:"comment"`
+	CreatedAt  *time.Time `json:"created_at"`
+	DeletedAt  *time.Time `json:"deleted_at"`
+}
+
+type TunnelRoutesListParams struct {
+	AccountID       string     `json:"-"`
+	TunnelID        string     `json:"tunnel_id,omitempty"`
+	Comment         string     `json:"comment,omitempty"`
+	IsDeleted       *bool      `json:"is_deleted,omitempty"`
+	NetworkSubset   string     `json:"network_subset,omitempty"`
+	NetworkSuperset string     `json:"network_superset,omitempty"`
+	ExistedAt       *time.Time `json:"existed_at,omitempty"`
+	PaginationOptions
+}
+
+type TunnelRoutesCreateParams struct {
+	AccountID string `json:"-"`
+	Network   string `json:"-"`
+	TunnelID  string `json:"tunnel_id"`
+	Comment   string `json:"comment,omitempty"`
+}
+
+type TunnelRoutesUpdateParams struct {
+	AccountID string `json:"-"`
+	Network   string `json:"network"`
+	TunnelID  string `json:"tunnel_id"`
+	Comment   string `json:"comment,omitempty"`
+}
+
+type TunnelRoutesForIPParams struct {
+	AccountID string `json:"-"`
+	Network   string `json:"-"`
+}
+
+type TunnelRoutesDeleteParams struct {
+	AccountID string `json:"-"`
+	Network   string `json:"-"`
+}
+
+// tunnelRouteListResponse is the API response for listing tunnel routes.
+type tunnelRouteListResponse struct {
+	Response
+	Result []TunnelRoute `json:"result"`
+}
+
+type tunnelRouteResponse struct {
+	Response
+	Result TunnelRoute `json:"result"`
+}
+
+// TunnelRoutes lists all defined routes for tunnels in the account.
+//
+// See: https://api.cloudflare.com/#tunnel-route-list-tunnel-routes
+func (api *API) TunnelRoutes(ctx context.Context, params TunnelRoutesListParams) ([]TunnelRoute, error) {
```

Comment:
shouldn't this have been called `ListTunnelRoutes()`?  I thought there was an effort to move all of the functions like this to be prefixed with `List`.

Reply:
we can however, i'm less concerned about this now that we have some progress on the upcoming [experimental](https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md) client that will do away with everything sitting on the same top level namespace.

we'll be adding that in pre-v2 and then dropping support for the older methods once v2 is released.

Reply:
I wasn't aware there was a new plan... so nevermind.

Reply:
I was simply going by #612 and #755, but with the new experimental bits it doesn't really matter, as everything will move to lower level functions.  And if Jacob isn't worried neither am I.